types/d3-sankey/index.d.ts

```diff
@@ -264,4 +342,4 @@ export function sankey(): SankeyLayout<{}, {}>;
  * Sankey layout generator. These properties are IN EXCESS to the properties explicitly identified in the
  * SankeyLinkMinimal interface.
  */
-export function sankey<N extends SankeyExtraProperties, L extends SankeyExtraProperties>(): SankeyLayout<N, L>;
+export function sankeyLinkHorizontal<N extends SankeyExtraProperties, L extends SankeyExtraProperties>(): Link<any, SankeyLink<N, L>, [number, number]>;
```

Comment:
Am I assuming correctly that you are referring to the `sankeyLinkHorizontal` signature? I thought about it in the context of `sankeyLinkHorizontal` being syntactic sugar for fully configuring your own horizontal  link using **d3-shape's** `linkHorizontal()`.

Given that it is already preconfigured with `source` and `target` accessors and assumes the default `x` and `y` accessors, it did not seem necessary to introduce a generic for the `this` context. As the preconfigured accessors do not access a specific `this` context. By the same token, I fixed the third generic of the returned `Link` to `[number, number]`. In principle, we could add additional generics/signatures, but it seems we would move away from the intended syntactic sugar, to a full horizontal link as **d3-shape** provides it.

Over to you, I will already push a commit addressing the above two comments in the meantime.